### DISCUSSION TITLE
Login State

### DISCUSSION THREAD

**Discussion by pamuji86** [mvanroon/react-native-axios-jwt#20] [Q&A]
Hi.
How to get current login state?
There's nothing to do with isLoggedIn function.

**Comment by mvanroon** [comment]
Not sure what you mean with ‘there is nothing to do’.

The isLoggedIn function that is being exported checks if the refresh tokens are present. If they are, that means that the user is authenticated (given that the tokens are valid)

**Comment by pamuji86** [comment]
I can get the stored token but that function can not set isLoggedIn as true.

**Comment by mvanroon** [comment]
When you store tokens using `setAuthTokens`, a user is considered authenticated.

`isLoggedIn` will return `true` If you have set the auth tokens and return `false` if you have not, so:

1. `isLoggedIn` is `false`
2. you log in with your backend and get auth tokens in return which you then store with `setAuthTokens`
3. when you call `isLoggedIn` again, the result is now `true`

I hope this helps.

**Comment by pamuji86** [comment]
`import React, { useState } from 'react';
import {
  isLoggedIn,
  setAuthTokens,
  clearAuthTokens,
  getAccessToken,
  getRefreshToken,
} from 'react-native-axios-jwt';
import { axiosInstance } from './api';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  Button,
  useColorScheme,
  View,
} from 'react-native';

import {
  Colors,
  DebugInstructions,
  Header,
  LearnMoreLinks,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

const App = () => {
if(!isLoggedIn()) { alert("Logged Out"); } else { alert("Logged In"); }
const logout = () => clearAuthTokens();
  const isDarkMode = useColorScheme() === 'dark';
  const [fullName, setFullName] = useState("");
  const [email, setEmail] = useState("");
  const onChangeNameHandler = (fullName) => {
    setFullName(fullName);
  };
   const onChangeEmailHandler = (email) => {
    setEmail(email);
  };  
  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.darker,
  };
	const login = async (params: ILoginRequest) => {
	  const response = await axiosInstance.post('/accounts/token/', {'username':''+fullName+'','password':''+email+''})
	  await setAuthTokens({
		accessToken: response.data.access,
		refreshToken: response.data.refresh
	  })
	}

const refreshToken = getRefreshToken().then(refreshToken => console.log(refreshToken))

if(!isLoggedIn() ) {
	return(
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <Header />
		<View><Button title="Logout" style={styles.button} onPress={() => logout()} /></View>
      </ScrollView>
    </SafeAreaView>	
	)} else {	
  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <Header />
		<View>
			<TextInput
          style={styles.input}
          placeholder="Email"
          placeholderTextColor="#fefefe"
          keyboardType="email-address"
          autoCapitalize="none"
		  // value={fullName}
          onChangeText={onChangeNameHandler}
          
        />

        <TextInput
          style={styles.input}
          placeholder="Password"
          placeholderTextColor="#fefefe"
          secureTextEntry
		  // value={email}
          onChangeText={onChangeEmailHandler}
          
        />
		<Button title="Login" style={styles.button} onPress={() => login()} />
		</View>
      </ScrollView>
    </SafeAreaView>
  );
  }
};


const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 14,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default App;`

Can't get it to work right now. Keep "LoggedIn" alert even in first run.

**Comment by mvanroon** [comment]
I'm sorry, but I can't read your code at all. Would you mind formatting your code properly and removing everything that has nothing to do with authentication?

Also keep in mind that isLoggedIn returns a promise.

So:

```
isLoggedIn().then(result => console.log(result))
```

or 

```
const isUserLoggedIn = await isLoggedIn()
console.log(isUserLoggedIn)
```
